src/dataset.rs

```diff
@@ -400,6 +400,13 @@ impl Dataset {
         Ok(Dataset { c_dataset })
     }
 
+    /// Flush all write cached data to disk.
+    ///
+    /// See [`GDALFlushCache`].
+    pub fn flush_cache(&self) {
```

Comment:
```suggestion
    pub fn flush_cache(&self) {
```

Let's take a hint from `std::fs::File` here.